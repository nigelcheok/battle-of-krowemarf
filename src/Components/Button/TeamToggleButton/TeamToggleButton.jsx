import React from 'react';

export function TeamToggleButton(props) {
  return (
    <div className="btn-group" role="group" aria-label="">
      { props.teams.map((team,index) => {
          return (
            <button
              key={index}
              type="button"
              className={`btn btn-outline-primary font-weight-bold ${props.currTeam === team ? 'active' : ''}`}
              onClick={ () => { props.onClanChange(team) } }
            >
             <div className={`fab ${team.clanIcon} fa-1x`} style={TeamToggleButtonIconStyle()}/>
             <div className="font-size-smaller display-inline-block">{ team.clanName }</div>
            </button>
          )
       })
     }
    </div>
  );
}

function TeamToggleButtonIconStyle() {
  return {
    display: 'inline-block',
    marginRight: '6px',
  }
}
