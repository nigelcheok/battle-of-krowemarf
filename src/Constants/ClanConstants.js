const allClans = [
  {
    clanName: 'React',
    clanIcon: 'fa-react',
    ownerName: 'facebook',
    repoName: 'react',
  },
  {
    clanName: 'Vue',
    clanIcon: 'fa-vuejs',
    ownerName: 'vuejs',
    repoName: 'vue',

  },
  {
    clanName: 'Angular',
    clanIcon: 'fa-angular',
    ownerName: 'angular',
    repoName: 'angular.js',
  }
];

export default {
  allClans
}