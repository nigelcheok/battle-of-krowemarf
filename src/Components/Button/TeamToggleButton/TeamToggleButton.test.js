import React from 'react';
import { shallow } from 'enzyme';

import { TeamToggleButton } from './TeamToggleButton';
import ClanConstants from "../../../Constants/ClanConstants";

const clickFn = jest.fn();

describe('TeamToggleButton', () => {
  it('should render correctly with no props', () => {
    const component = shallow(<TeamToggleButton/>);
    expect(component).toMatchSnapshot();
  });

  it('should render correctly with teams and currteam', () => {
    const component = shallow(<TeamToggleButton teams={ClanConstants.allClans} currTeam={ClanConstants.allClans[1]} onClanChange={clickFn}/>);

    component.find('button#team-Angular')
      .simulate('click');

    expect(clickFn).toHaveBeenCalled();
    expect(component).toMatchSnapshot();
  });
});