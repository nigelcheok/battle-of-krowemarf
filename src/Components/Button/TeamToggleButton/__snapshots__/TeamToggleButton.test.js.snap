// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamToggleButton should render correctly with no props 1`] = `
<div
  aria-label=""
  className="btn-group"
  role="group"
/>
`;

exports[`TeamToggleButton should render correctly with teams and currteam 1`] = `
<div
  aria-label=""
  className="btn-group"
  role="group"
>
  <button
    className="btn btn-outline-primary font-weight-bold "
    id="team-React"
    key="0"
    onClick={[Function]}
    type="button"
  >
    <div
      className="fab fa-react fa-1x"
      style={
        Object {
          "display": "inline-block",
          "marginRight": "6px",
        }
      }
    />
    <div
      className="font-size-smaller display-inline-block"
    >
      React
    </div>
  </button>
  <button
    className="btn btn-outline-primary font-weight-bold active"
    id="team-Vue"
    key="1"
    onClick={[Function]}
    type="button"
  >
    <div
      className="fab fa-vuejs fa-1x"
      style={
        Object {
          "display": "inline-block",
          "marginRight": "6px",
        }
      }
    />
    <div
      className="font-size-smaller display-inline-block"
    >
      Vue
    </div>
  </button>
  <button
    className="btn btn-outline-primary font-weight-bold "
    id="team-Angular"
    key="2"
    onClick={[Function]}
    type="button"
  >
    <div
      className="fab fa-angular fa-1x"
      style={
        Object {
          "display": "inline-block",
          "marginRight": "6px",
        }
      }
    />
    <div
      className="font-size-smaller display-inline-block"
    >
      Angular
    </div>
  </button>
</div>
`;
